### ISSUE TITLE
Can vrvToolkit.getElementAttr() be made to work on the "whole" MEI?

### ISSUE THREAD

**Issue by musicog** [rism-ch/verovio#1124]
**Problem:**
In Python, calling vrvTk.getElementAttr() on an xml ID returns an empty object instead of the element attributes as expected. 

**Verovio versions affected:**
Reproduced in master (`vrvTk.getVersion() == '2.1.0-824126c'`) and develop (`'2.2.0-dev-15fc843'`) branches

**Repro steps:**
Using MEI file from https://www.verovio.org/examples/downloads/Brahms_StringQuartet_Op51_No1.mei

```
>>> import verovio
>>> vrvToolkit = verovio.toolkit()
>>> vrvToolkit.loadFile("Brahms_StringQuartet_Op51_No1.mei")
True
>>> vrvToolkit.getElementAttr("d648110e110")
'{\n} \n'
```
For comparison, in Javascript it works as expected:

```
vrvToolkit.getElementAttr("d648110e110")
> Object { dur: "8", oct: "4", pname: "e", "stem.dir": "up" }
```

**Comment by musicog** [comment]
Although, from further experimentation, it seems in JS it only works for some IDs:

```
vrvToolkit.getElementAttr('d648110e5628')
> Object {  }
```

From the MEI:

```
<note xml:id="d648110e110" pname="e" oct="4" dur="8" stem.dir="up" accid.ges="f"/>
<note xml:id="d648110e5628" pname="a" oct="4" dur="4" stem.dir="up" accid.ges="f"/>

```
The first one works (produces element attributes), the second one breaks (produces an empty object) using the Javascript toolkit as currently deployed on the Verovio website - version `2.2.0-dev-c01b3a1-dirty`.

Neither (none I've tried so far) works in Python.

**Comment by musicog** [comment]
... and after further experimentation :-)

I now understand that the note has to be rendered into the current DOM for the JS version to work, which explains why the Python version doesn't work -- no DOM!

Is there a way to expose this information from the MEI directly through Verovio, rather than (what I assume is happening) from the currently generated SVG?

**Comment by vitto32** [comment]
I've noticed a similar problem with 2.2.0-dev-c01b3a1-dirty
 `vrv.renderToSVG(page, {});
vrv.getElementAttr(svg-id)
`

**Comment by musicog** [comment]
Also if zoom is set to very tiny so that everything fits in one page. In any case, the whole file needs to be rendered at once for every ID to work.

@vitto32 Could you confirm: in 2.0 it worked even for notes not-currently-rendered?

**Comment by vitto32** [comment]
@musicog in 2.0 it only works for notes in the most recently rendered page. By rendered I mean that renderToSVG is called and a string is returned (no DOM involved).

In 2.2 it does not work for currently rendered page (except if page contains full score)

**Comment by musicog** [comment]
@vitto32 is right, my intuition about the DOM is mistaken, all that seems to matter is that `renderToSVG` is called first. In Python:

```
>>> import verovio
>>> vrvTk = verovio.toolkit()
>>> vrvTk.loadFile('example.mei')
[Message] Converting analytical markup...
True
>>> vrvTk.getElementAttr('note-0000000119483089')
'{\n} \n'
>>> svg = vrvTk.renderToSVG()
>>> vrvTk.getElementAttr('note-0000000119483089')
'{\n\t"dur": "16",\n\t"oct": "2",\n\t"pname": "a",\n\t"stem.dir": "up" \n} \n'

```

**Comment by lpugin** [comment]
Should be fixed with 1cb8304df79900ed5d04c61834719119bdb885ec and no rendering beforehand should be required at all.

**Comment by vitto32** [comment]
@lpugin I can confirm it's fixed
